tests/cpp_api_tests/rocAL_audio_unittests/rocAL_audio_unittests.cpp

```diff
@@ -45,35 +45,50 @@ bool verify_output(float *dst_ptr, long int frames, std::string case_name) {
     }
 
     std::string ref_file_path = std::string(rocal_data_path) + "GoldenOutputsTensor/reference_outputs_audio/" + case_name + "_output.bin";
-    long int out_buffer_size = frames;
-    std::vector<float> ref_output(out_buffer_size);
-    std::fstream fin(ref_file_path, std::ios::in | std::ios::binary);
-    if (fin.is_open()) {
-        for (long int i = 0; i < out_buffer_size; i++) {
-            if (!fin.eof()) {
-                fin.read(reinterpret_cast<char *>(ref_output.data()), sizeof(float));
-            } else {
-                std::cout << "\nUnable to read all data from golden outputs\n";
-                return pass_status;
-            }
-        }
-    } else {
-        std::cout << "\nCould not open the reference output. Please check the path specified\n";
-        return pass_status;
+    std::ifstream fin(ref_file_path, std::ios::binary); // Open the binary file for reading
+
+    if (!fin.is_open()) {
+        std::cerr << "Error: Unable to open the input binary file\n";
+        return 1;
+    }
+
+    // Get the size of the file
+    fin.seekg(0, std::ios::end);
+    std::streampos fileSize = fin.tellg();
+    fin.seekg(0, std::ios::beg);
+
+    // Calculate the number of floats in the file
+    std::size_t numFloats = fileSize / sizeof(float);
+
+    // Create a vector to store the floats
+    std::vector<float> ref_output(numFloats);
+
+    // Read the floats from the file
+    fin.read(reinterpret_cast<char*>(ref_output.data()), fileSize);
```

Comment:
```suggestion
    // Calculate the number of floats in the file
    std::size_t numFloats = fileSize / sizeof(float);

    // Create a vector to store the floats
    std::vector<float> ref_output(numFloats);

    // Read the floats from the file
    fin.read(reinterpret_cast<char*>(ref_output.data()), fileSize);
```

We need not add description here to every line I think it is implied, Only the last line's comment can be retained